### PR TITLE
Misra 14 4 fixes

### PR DESCRIPTION
**Pull request by ceolin**
Make statements evaluate boolean expressions. These commits are manual changes. I've another pr with automatic changes using coccinelle.

### CODE REVIEW

File: kernel/include/syscall_handler.h [hunk 1]
```diff
@@ -406,12 +406,12 @@ static inline int _obj_validation_check(struct _k_object *ko,
 	ARG_UNUSED(obj);
 #endif
 
-	return ret;
+	return ret == 0 ? true : false;
```
Comment (ioannisg) [inline kernel/include/syscall_handler.h]: do we need the '?' here?

Comment (ioannisg) [reply kernel/include/syscall_handler.h]: Could we have `return ret !=0`?

File: kernel/include/syscall_handler.h [hunk 2]
```diff
@@ -406,12 +406,12 @@ static inline int _obj_validation_check(struct _k_object *ko,
...
 }
 
 #define Z_SYSCALL_IS_OBJ(ptr, type, init) \
 	Z_SYSCALL_VERIFY_MSG( \
-	    !_obj_validation_check(_k_object_find((void *)ptr), (void *)ptr, \
+	    _obj_validation_check(_k_object_find((void *)ptr), (void *)ptr, \
```
Comment (ioannisg) [inline kernel/include/syscall_handler.h]: that's a bit hard to read, now. I wonder if the function name could also be updated to reflect the revert of the return value

**Review by ioannisg — COMMENTED** [review]
Generally looks ok to me; left some comments.

File: include/arch/arc/syscall.h [hunk 3]
```diff
@@ -182,7 +183,7 @@ static inline int _arch_is_user_context(void)
 			 : "=r"(status)
 			 : [status32] "i" (_ARC_V2_STATUS32));
 
-	return !(status & _ARC_V2_STATUS32_US);
+	return (status & _ARC_V2_STATUS32_US) ? true : false;
```
Comment (ruuddw) [inline include/arch/arc/syscall.h]: I believe you missed the ! on the original line, please swap true and false (and, would be good to check effiency of generated code to ensure no performance penalty).

**Review by ruuddw — DISMISSED** [review]

File: kernel/include/syscall_handler.h (hunk 1 above)
Comment (ceolin) [reply kernel/include/syscall_handler.h]: yeah, we can, it's better indeed.

File: include/arch/arc/syscall.h (hunk 3 above)
Comment (ceolin) [reply include/arch/arc/syscall.h]: yeah, thanks for pointing it out.

**Review by tyrantlizard77 — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**